runtime/minor_gc.c

```diff
@@ -166,25 +166,28 @@ static inline void log_gc_value(const char* prefix, value v)
 }
 #endif
 
-#define Color_hd(v) ((color_t)(((v) >> 8) & 3))
-
-int get_header_val(value v) {
-  if( caml_domain_alone() ) {
-    return Hd_val(v);
-  } else {
-    header_t hd = atomic_load_explicit(Hp_atomic_val(v), memory_order_relaxed);
-
-    if ( Wosize_hd(hd) == 0 && Color_hd(hd) == 1 ) {
-      while( atomic_load_explicit(Hp_atomic_val(v), memory_order_acquire) != 0 ) {}
-      return 0;
-    }
+/* in progress updates are zeros except for the lowest color bit set to 1 */
+#define In_progress_update_val ((header_t)0x100)
```

Comment:
I would add a comment on this line to explain how this maps to the old code (i.e Wosize_hd(hd) == 0 && Color_hd(hd) == 1)